### PR TITLE
Adding Exchange auth method

### PR DESCRIPTION
**Pull request by georific**
## Status
**READY**

## JIRA Ticket
MOB-203

## Description
Add ability to auth directly to exchange endpoint.

## Todos
- [ ] Tests
- [ ] Documentation
- [ ] License

## Steps to Test or Reproduce
Outline the steps to test or reproduce the PR here.

```sh
git fetch --all
git checkout <feature_branch> 
./gradlew connectedCheck
```

@boundlessgeo/spatial-connect

### CODE REVIEW

File: spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/SpatialConnect.java [hunk 1]
```diff
@@ -47,7 +47,7 @@
 public class SpatialConnect {
 
     private final String LOG_TAG = SpatialConnect.class.getSimpleName();
-    private SCServiceGraph serviceGraph;
+    private SCServiceGraph serviceGraph = new SCServiceGraph();;
```
Comment (mrcnc) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/SpatialConnect.java]: Remove the duplicate semicolon here

File: spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/authService/SCExchangeAuthMethod.java [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        }
+    }
+
+    @Override
+    public boolean authenticate(String username, String pwd) {
+        return auth(username, pwd);
+    }
+
+    @Override
+    public boolean refreshToken() {
```
Comment (mrcnc) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/authService/SCExchangeAuthMethod.java]: Is this ever called anywhere?  I couldn't find it.

Comment (georific) [reply spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/authService/SCExchangeAuthMethod.java]: Not yet, the next PR will actually call it,  will need to check everytime it attempts to auth against the cache, and also need to figure out a good way to check on every http request, there could be an instance when the app is open and the token has expired.

File: spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/SpatialConnect.java (hunk 1 above)
Comment (georific) [reply spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/SpatialConnect.java]: good catch

File: spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/authService/SCExchangeAuthMethod.java [hunk 3]
```diff
@@ -110,23 +131,29 @@ private boolean auth(final String username, final String pwd) {
...
             final String oauthCreds = String.format(Locale.US, "%s:", clientId);
             final String base64Encoded = Base64.encodeToString(oauthCreds.getBytes("UTF-8"), Base64.NO_WRAP);
             final String authHeader = String.format(Locale.US, "Basic %s", base64Encoded);
             Response response = HttpHandler.getInstance()
                     .postBlocking(theUrl,
                             String.format("grant_type=refresh_token&refresh_token=%s",
-                                    getAccessToken()), authHeader, HttpHandler.XML);
+                                    cache.getStringValue(REFRESH_TOKEN)), authHeader, HttpHandler.XML);
 
+            JSONObject responseJson = new JSONObject(response.body().string());;
```
Comment (mrcnc) [inline spatialconnect/src/main/java/com/boundlessgeo/spatialconnect/services/authService/SCExchangeAuthMethod.java]: Remove the duplicate semicolon please

**Comment by mrcnc** [comment]
👍

**Pull request merged**